### PR TITLE
Feat sort new

### PR DESCRIPTION
**Pull request by MertsAnastasiya**

### CODE REVIEW

File: src/modules/filters/globalFilters.ts [hunk 1]
```diff
@@ -11,35 +11,44 @@ import {
 import { IProduct } from '../interfaces/product.interface';
 import { DualSlider } from './sliders';
 import { Search } from './search';
+import { Sort } from './sort';
 
 //I'll change it later when all layout is refactored
 const wrapperFiltres: Element = document.querySelector('.filters')!;
 const wrapperSliders: Element = document.querySelector('.sliders')!;
+const WRAPPER_SORT: Element = document.querySelector('.sort__wrapper')!;
```
Comment (alexey-ulashchick) [inline src/modules/filters/globalFilters.ts:19]: Why is this capitalized suddenly?

File: src/modules/filters/result.ts [hunk 2]
```diff
@@ -51,7 +52,30 @@ export class FilterResult {
...
+                : this.sortByProperty(productsResult, sortData[0]! as keyof IProduct).reverse();
+
+        }
         return productsResult;
     }
+
+    private static sortByProperty(productsData: IProduct[], sortProperty: keyof IProduct): IProduct[] {
+        return productsData.sort((a, b) => {
+            const nameA = a[sortProperty];
+            const nameB = b[sortProperty];
```
Comment (alexey-ulashchick) [inline src/modules/filters/result.ts:71]: Could you explicit add types here for `a`, `b`, `nameA` and `nameB`. 
Comparison is unclear for the reader.

File: src/modules/filters/sort.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { onSortClick } from '../interfaces/customTypes';
+
+export class Sort {
```
Comment (alexey-ulashchick) [inline src/modules/filters/sort.ts:3]: I would avoid to name anything with just pure word "Sort". Why it's not a keyword, it's confusing to have general "Sort". I would suggest to add some specifics.

File: src/modules/filters/sort.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        sort.classList.add('sort');
+        sort.classList.add(`sort__${this.sortProperty}`);
+        sort.innerText = this.sortProperty;
+
+        sort.addEventListener('click', () => {
+            if (this.sortDirection === 'none' || this.sortDirection === 'asc') {
+                this.sortDirection = 'desc';
+            } else if (this.sortDirection === 'desc') {
+                this.sortDirection = 'asc';
+            }
```
Comment (alexey-ulashchick) [inline src/modules/filters/sort.ts:27]: I think, you should have `else` fallback here. Either thrown an error with unrecognized `sortDirection` or rework this code part to make sure that `sortDirection` is always correct.

File: src/modules/interfaces/customTypes.ts [hunk 5]
```diff
@@ -10,7 +10,8 @@ export type CallbackOnClickCheckbox = (filterType: FilterType, value: string, is
 
 export type CallbackViewChanged = (sliderType: SliderType, currentSliderValue: SliderValue) => void;
 
-export type CallbackOnChangeFilters = (currentFilters: Map<FilterType, Set<string>>, currentSliders: Map<SliderType, SliderValue>, searchValue: string) => void;
+export type CallbackOnChangeFilters = (currentFilters: Map<FilterType, Set<string>>, currentSliders: Map<SliderType, SliderValue>, searchValue: string, sort: string) => void;
+// export type CallbackOnChangeFilters = (currentFilters: Map<FilterType, Set<string>>, currentSliders: Map<SliderType, SliderValue>, searchValue: string) => void;
```
Comment (alexey-ulashchick) [inline src/modules/interfaces/customTypes.ts:14]: Dead code no explanation.

**Review by alexey-ulashchick — CHANGES_REQUESTED** [review]

File: src/modules/filters/globalFilters.ts (hunk 1 above)
Comment (MertsAnastasiya) [reply src/modules/filters/globalFilters.ts:19]: Good question)
I used upper-case in app.ts and then decided to use it here as well. Do you think it makes sense to use upper-case for constants?

File: src/modules/filters/result.ts (hunk 2 above)
Comment (MertsAnastasiya) [reply src/modules/filters/result.ts:71]: Added.

File: src/modules/filters/sort.ts (hunk 3 above)
Comment (MertsAnastasiya) [reply src/modules/filters/sort.ts:3]: Done.

File: src/modules/filters/sort.ts (hunk 4 above)
Comment (MertsAnastasiya) [reply src/modules/filters/sort.ts:27]: Fixed.

File: src/modules/interfaces/customTypes.ts (hunk 5 above)
Comment (MertsAnastasiya) [reply src/modules/interfaces/customTypes.ts:14]: Fixed.

File: src/modules/filters/globalFilters.ts (hunk 1 above)
Comment (alexey-ulashchick) [reply src/modules/filters/globalFilters.ts]: It's up to you. Better check with style guides. But anyway: either all capitalized or all camelCase.

**Review by alexey-ulashchick — APPROVED** [review]

**Pull request merged**